servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingHttpService.java

```diff
@@ -46,4 +52,94 @@
     default Completable closeAsync() {
         return completed();
     }
+
+    /**
+     * Wraps the passed {@link StreamingHttpService} to apply the provided {@link HttpExecutionStrategy} for offloading.
+     *
+     * @param strategy {@link HttpExecutionStrategy} to use for offloading.
+     * @param executor {@link Executor} to use as executor or {@code null} to use Execution context executor.
+     * @param shouldOffload If {@link BooleanSupplier} returns {@code true} then offload to executor otherwise continue
+     * execution on calling thread.
+     * @param service {@link StreamingHttpService} to wrap.
+     * @return Wrapped {@link StreamingHttpService}.
+     */
+    static StreamingHttpService offloadService(final HttpExecutionStrategy strategy,
```

Comment:
lets move this into a different utility file. most users aren't expected to know this exists and don't need to know about it.

Reply:
Now in `StreamingHttpServiceToOffloadedStreamingHttpService`. The naming matches the naming of the other adapters to blocking and aggregating.